lib/membrane/rtp/session_bin.ex

```diff
@@ -253,61 +253,35 @@ defmodule Membrane.RTP.SessionBin do
   end
 
   @impl true
-  def handle_pad_added(Pad.ref(:rtp_input, ref) = pad, ctx, %{secure?: true} = state) do
+  def handle_pad_added(Pad.ref(:rtp_input, ref) = pad, ctx, %{secure?: secure?} = state) do
     rtcp_output = Pad.ref(:rtcp_output, ref)
     rtcp? = Map.has_key?(ctx.pads, rtcp_output)
 
-    links =
-      [
-        link_bin_input(pad, buffer: @rtp_input_buffer_params)
-        |> to({:rtp_parser, ref}, %RTP.Parser{secure?: true})
-        |> via_out(:output)
-        |> via_in(Pad.ref(:input, ref))
-        |> to(:ssrc_router)
-      ] ++
-        if rtcp? do
-          [
-            link({:rtp_parser, ref})
-            |> via_out(:rtcp_output)
-            |> to({:rtcp_decryptor, ref}, %Membrane.SRTCP.Decryptor{policies: state.srtp_policies})
-            |> to({:rtcp_parser, ref}, RTCP.Parser)
-            |> via_out(:rtcp_output)
-            |> to({:srtcp_encryptor, ref}, %SRTP.Encryptor{policies: state.receiver_srtp_policies})
-            |> to_bin_output(rtcp_output),
-            link({:rtcp_parser, ref})
-            |> via_out(:output)
-            |> via_in(Pad.ref(:input, {:rtcp, ref}))
-            |> to(:ssrc_router)
-          ]
-        else
-          []
-        end
+    maybe_link_rtcp_decryptor =
+      &to(&1, {:rtcp_decryptor, ref}, %Membrane.SRTCP.Decryptor{policies: state.srtp_policies})
 
-    {{:ok, spec: %ParentSpec{links: links}}, state}
-  end
-
-  @impl true
-  def handle_pad_added(Pad.ref(:rtp_input, ref) = pad, ctx, state) do
-    rtcp_output = Pad.ref(:rtcp_output, ref)
-    rtcp? = Map.has_key?(ctx.pads, rtcp_output)
+    maybe_link_rtcp_encryptor =
+      &to(&1, {:srtcp_encryptor, ref}, %Membrane.SRTP.Encryptor{
```

Comment:
either `rtcp_encryptor` and `rtcp_decryptor` or `srtcp_encryptor` and `srtcp_decryptor`